pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -676,30 +590,20 @@ def _set_snopt_options(self, iPrint, iSumm, cw, iw, rw):
         Set all the options into SNOPT that have been assigned
         by the user
         """
-
         # Set Options from the local options dictionary
         # ---------------------------------------------
         inform = np.array([-1], np.intc)
-        for name, value in self.set_options.items():
-            if name == "iprint" or name == "isumm":
+        for name, value in self.options.items():
+            # these do not get set using snset
+            if name in self.specialOptions or name in self.pythonOptions:
```

Comment:
Are you sure you are not skipping stuff here? it seems like you skip the `specialOptions` with `continue` but then you are trying to use some of them (`Print file`) to send stuff to SNOPT via `snset`

Reply:
Yes I made a mistake here. Those two should not be in `specialOptions`, but also I had to adjust the capitalization for it to work with SNOPT. I ended up adding a `caseSensitiveOptions` param to BaseSolver to force all options to be case sensitive since some optimizers expect it to be, and this was the previous behaviour of the Optimizer class.

Reply:
Cool, defaulting to lower case and using the `caseSensitiveOptions` flag only when needed makes sense to me, still support the choice you made for `baseclasses`